### PR TITLE
Firing pin additions

### PR DESCRIPTION
**Pull request by Hackole**
Adds a syndicate firing pin, which will deny firing whilst at the syndicate outpost. This will hopefully cut down on nuke-teams blowing themselves up before they even get to the station. The pin has been added to most guns that they can get their hands on.

Adds a trick firing pin to uplinks for 3tc. Upon trying to fire a weapon with this pin installed, it will explode, likely critting whoever the victim is.

### CODE REVIEW

**Comment by trifornt-tinfornt** [comment]
http://i.imgur.com/xcjS4MP.png
it sucks
On a more serious note, it's a godsend. No more idiotic leaders declaring war and buying blastco without asking their team which always leads to the inevitable newbie Op misfiring the Striders and killing everyone.
Trick Pins sound like fun, though they should have tacticool pin replacement where if you simply place it in a gun, pin or not, it'll automatically place its self in instead.

**Comment by myontasm** [comment]
came in here expecting shitty firing pin additions (as the whole firing pin thing is unused and thus useless)

this is very good I like it

**Comment by ghost** [comment]
Good shit +1

**Comment by ArcLumin** [comment]
Excellent. I like it

**Comment by Gandalf2k15** [comment]
+19999

**Comment by dreamcrab** [comment]
this is creative and cool +420

**Comment by Voratius** [comment]
Sure, sounds like a neato featuro +1/3rd.

Have you playtested it completely to make sure there aren't any issues?

**Comment by Carbonhell** [comment]
god no,the idea behind it is ok but the way it's implemented isn't really good,i'd have preferred if it checked the area it's in, aka make an area setting that disallows any kind of fight or idk

File: code/modules/projectiles/guns/projectile/revolver.dm [hunk 1]
```diff
@@ -102,6 +103,7 @@
 	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev38
 	unique_rename = 1
 	unique_reskin = 1
+	pin = /obj/item/device/firing_pin/generic
```
Comment (JohnGinnane) [inline code/modules/projectiles/guns/projectile/revolver.dm:106]: How come these guns get generic pins?

File: code/modules/projectiles/pins.dm [hunk 2]
```diff
@@ -112,6 +112,32 @@
...
+/obj/item/device/firing_pin/area/syndicate
+	force_replace = 1
+	A = /area/syndicate_mothership
+	name = "Syndicate firing pin"
+	desc = "This pin will prevent firing whilst at the operative outpost."
+	icon_state = "firing_pin_pindi"
+
+/obj/item/device/firing_pin/trick
+	name = "Generic firing pin"
+	desc = "This pin will detonate the weapon it is put into upon trying to use it"
```
Comment (JohnGinnane) [inline code/modules/projectiles/pins.dm:135]: > ..detonate the weapon it is put into upon trying..

I'm not sure I understand what this is trying to say?

**Comment by JohnGinnane** [comment]
I kinda disagree Carbon. This is pretty straightforward and doesn't require changing areas. It also means you could add a pin to stop some mining equipment being used outside the mining area potentially, again without having to change area definitions too much

**Comment by Spacedong** [comment]
pretty good and I can see it being used for other stuff too maybe in the future, just make sure grenade launchers have them, thats the most important thing

File: code/modules/projectiles/pins.dm (hunk 2 above)
Comment (Hackole) [inline code/modules/projectiles/pins.dm:135]: It's saying it detonates the weapon when it is fired with the pin in - bad wording on my part.

File: code/modules/projectiles/guns/projectile/revolver.dm (hunk 1 above)
Comment (Hackole) [inline code/modules/projectiles/guns/projectile/revolver.dm:106]: The parent revolver was the syndicate variety available to nukeops, so I had to edit it to have the syndicate pin. It wouldn't make sense for the other ones to have syndicate pins too, so I just made them have generic ones.

**Comment by Hackole** [comment]
Yes spacedong, I ensured that the grenade launchers had the pins. That was the biggest offender.

**Comment by Kokojos** [comment]
This has reached the votes, and code seems fine.

**Pull request merged**